engines/freescape/detection.cpp

```diff
@@ -0,0 +1,248 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "base/plugins.h"
+#include "engines/advancedDetector.h"
+#include "common/translation.h"
+
+#include "freescape/freescape.h"
+
+#define GAMEOPTION_PRERECORDED_SOUNDS   GUIO_GAMEOPTIONS1
+
+namespace Freescape {
+
+static const PlainGameDescriptor freescapeGames[] = {
+	{"3dkit", "The 3D Kit Game"},
+	{"driller", "Driller"},
+	{"spacestationoblivion", "Space Station Oblivion"},
+	{"darkside", "Dark Side"},
+	{"totaleclipse", "Total Eclipse"},
+	{"castlemaster", "Castle Master"},
+	{"menace", "Menace of Dr. Spoil Sport"},
+	{0, 0}};
+
+static const ADGameDescription gameDescriptions[] = {
+	{"driller",
+	 "Driller",
+	 {
+		{"DRILLER.EXE", 0, "cafc0ea0d3424640a7723af87f8bfc0b", 17427},
```

Comment:
Yes, these games are somehow unusual. DRILLE.exe is the EGA game executable, DRILLC is the CGA game executable and so on. All these include full copy of the game data (the data is actually different from depending on the number of colors, and with some other minor differences). However, these different executable were never distributed separately, but I believe some releases lack of some of these.

Only in Castle Master and some versions of Driller, the executable code and the data is actually separated (well, part of it, at least).

Reply:
I still do not understand. Detection tables are meant to identify unique games, not checking that all files are present etc. Are there other games with the same checksums and size in DRILLER.EXE or any other file from this table?

Reply:
Uhm, probably in the case of Driller, only DRILLER.EXE could be used (if there is only one release, which we don't know), but in the case of Castle Master, more files than CASTLE.EXE could be necessary. There are also pirated releases, which are not included here of course, and could require additional files to be detected.

Reply:
Please, include a minimal number of files in each entry.

As for the pirated versions, it is best to include them and mark them with `ADGF_PIRATED`. Otherwise, you will get them reported as unknown md5s.